language-server/dm/rawDocument.ml

```diff
@@ -87,7 +87,7 @@ let range_of_loc raw loc =
   }
 
 let word_at_position raw pos : string option =
-  let r = Str.regexp {|\([a-zA-Z_][a-zA-Z_0-9]*\)|} in
+  let r = Str.regexp {|\([a-zA-Z_.][a-zA-Z_0-9.']*\)|} in
```

Comment:
I don't get why the . need to be part of the first character set.
The code below tries the regexp at a position and then goes back until it can.

Reply:
I originally didn't have it there but encountered the following issue since the code rolls back a single character at a time:
A case like `Nat.pl|us` it would roll back to `Nat|.plus` which then would no longer match unless the . was in first character set.

If you want I think a more robust way would be to roll back as far as possible to until you encounter `[^a-zA-Z_0-9.']` then consume forward with `{|\([a-zA-Z_][a-zA-Z_0-9.']*\)|}`

Reply:
then there is an off by one error in the previous code

Reply:
Hmm yes I do think some more errors could creep up. If you hover something like `lemma1_te|st` I have a feeling it would stop at `lemma|1_test` since `[0-9]` does not appear in the first character set

Reply:
Just confirmed this is the case, will rework to be more robust